Add Logstash 6.x compatibility

robbavey opened the pull request:
This commit adds:
  Support for Logtash 6.x
  Travis integration preparation
  Reworks changelog to be 'newest first'

robbavey committed:
Add Logstash 6.x compatibility

This commit adds:
  Support for Logtash 6.x
  Travis integration preparation
  Reworks changelog to be 'newest first'

robbavey committed:
Adds a unit test

Adds the standard "behaves like an interruptible input plugin" test

jsvd commented on lib/logstash/inputs/s3sqs.rb:
```diff
@@ -6,6 +6,9 @@
 require "logstash/errors"
 
+# https://aws.amazon.com/blogs/developer/threading-with-the-aws-sdk-for-ruby/.
+# Aws.eager_autoload!
```
is this necessary?

robbavey replied:
I think so - I saw some weird dependency issues during testing, which didn't happen when I reinstated this line.

jsvd reviewed (approved):
code LGTM, minor nitpick on commented code

robbavey committed:
Add call to eager_autoload!

During testing on 5.x versions of Logstash, I encountered some weird
dependency errors, which appear to be related to the issues
described in https://aws.amazon.com/blogs/developer/threading-with-the-aws-sdk-for-ruby/
which can occur in versions of Jruby predating jruby-9k.

This commit adds the call to eager_autoload! as recommended in the
article above
